### PR TITLE
Feat/improve windows install script

### PR DESCRIPTION
**Pull request by brittandeyoung**
Added optional parameters to improve the install script and updated the documentation to reflect this update:

* `-DaggerVersion`: allows specifying the version to install. Will default to `latest`
* `-InteractiveInstall`: Allows running the script unattended. Defaults to `$false`
* `-InstallPath`: Allows passing in a specific location for installing the `dagger.exe` binary.

### CODE REVIEW

**Comment by grouville** [comment]
@brittandeyoung, thanks for the rebase. The windows computer is at home. I'll check it later today 😇

File: install.ps1
Comment (jpadams) [inline install.ps1:3]: I get this error when I run `install.ps1`:

```
Unable to find type [System.Management.Automation.SemanticVersion].
At C:\Users\Home\src\dagger\install.ps1:3 char:37
+ ... ory = $false)] [System.Management.Automation.SemanticVersion]$DaggerV ...
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (System.Manageme...SemanticVersion:TypeName) [], RuntimeException
+ FullyQualifiedErrorId : TypeNotFound
```

Comment (jpadams) [reply install.ps1:3]: PS C:\Users\Home\src\dagger> $PSVersionTable
>>

Name                           Value
----                           -----
PSVersion                      5.1.22000.653
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.22000.653
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

Comment (jpadams) [reply install.ps1:3]: Looks like I need to move to PS 7.x. 😄 
https://github.com/PowerShell/PowerShell/issues/16330

Comment (jpadams) [reply install.ps1:3]: There are likely other folks on old Powershell versions. Can we test for and require a minimum PS version in the script so folks know what to do?

Comment (jpadams) [reply install.ps1:3]: Trying this: `winget install Microsoft.PowerShell`

Screen Shot 2022-08-12 at 1 41 45 PM

Comment (jpadams) [reply install.ps1:3]: image

`-Interactive $true` asked me to proceed, but default it's set to `$false` so just installs. Could easily be flipped if important, so 👍 
image

Version and location settings work perfectly.

Comment (jpadams) [reply install.ps1:3]: So the summary is, works great, but we need to let folks know to be on `pwsh` aka version 7.x.
Thanks for this contribution @brittandeyoung 
It would be ideal to add requirement of `pwsh` version to the doc for sure, and to the script if possible.

Comment (brittandeyoung) [reply install.ps1:5]: Powershell version requirement added to the script and docs.

**Comment by brittandeyoung** [comment]
Rebase with suggested improvements

**Comment by brittandeyoung** [comment]
Bump @grouville

**Review by TomChv — COMMENTED** [review]
Code and explanation looks good.

@grouville Is there anything that block the merge right now?

**Comment by grouville** [comment]
# Tests:

Powershell 5.1 -> requirements now work:
image

Upgrading to 7:
image

Works fine ! Rebasing directly on your branch @brittandeyoung

**Review by grouville — APPROVED** [review]
LGTM

**Pull request merged**
